switchtec.c

```diff
@@ -1321,6 +1341,9 @@ static int mask_event(struct switchtec_dev *stdev, int eid, int idx)
 	hdr_reg = event_regs[eid].map_reg(stdev, off, idx);
 	hdr = ioread32(hdr_reg);
 
+	if (!event_supported(stdev, eid, idx))
```

Comment:
This seems overkill. We just read hdr on the line earlier and we read it again inside event_supported. Seems like you can just check the bit directly here.

Reply:
Will update.